File: src/main/java/com/patrickgrimard/examples/Item.java
package com.patrickgrimard.examples;

import javax.persistence.Entity;
import javax.persistence.Id;

/**
 * Created on 2016-11-25
 *
 * @author Patrick
 */
@Entity
public class Item {

    @Id
    private String name;

    public Item(String name) {
        this.name = name;
    }

    public Item() {

    }

    public String getName() {
        return name;
    }
}

File: src/main/java/com/patrickgrimard/examples/Item.kt
package com.patrickgrimard.examples

import javax.persistence.Entity
import javax.persistence.Id

/**
 *
 *
 * Created on 2016-12-03
 *
 * @author Patrick
 */
@Entity
data class Item(@Id val name: String, val quantity: Long = 0) {
    private constructor() : this("", 0)
}